builtin/logical/pki/backend.go

```diff
@@ -267,7 +267,7 @@ func (b *backend) metricsWrap(callType string, roleMode int, ofunc roleOperation
 			if err != nil {
 				return nil, err
 			}
-			if role == nil && roleMode == roleRequired {
+			if role == nil && (roleMode == roleRequired || role == roleOptional) {
```

Comment:
shouldn't this read more like this? As now I *believe* if we don't specify a role it will fail?
```suggestion
			if role == nil && (roleMode == roleRequired || len(roleName) > 0 {
```